okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -93,6 +98,21 @@
   private static final Logger log = Logger.getLogger(OkHttpClientTransport.class.getName());
   private static final OkHttpClientStream[] EMPTY_STREAM_ARRAY = new OkHttpClientStream[0];
 
+  private static final Resource<ExecutorService> SHARED_EXECUTOR =
+      new Resource<ExecutorService>() {
+        @Override
+        public ExecutorService create() {
+          return Executors.newCachedThreadPool(new ThreadFactoryBuilder()
```

Comment:
Just moving this code, not changing it ... we can always tackle in a separate PR if we want.